esiclient/v1/trunk.py

```diff
@@ -113,6 +114,125 @@ def take_action(self, parsed_args):
              trunk.sub_ports]
 
 
+class AddNetwork(command.ShowOne):
+    """Add tagged network"""
+
+    log = logging.getLogger(__name__ + ".AddNetwork")
+
+    def get_parser(self, prog_name):
+        parser = super(AddNetwork, self).get_parser(prog_name)
+        parser.add_argument(
+            "name",
+            metavar="<name>",
+            help=_("Name of trunk"))
+        parser.add_argument(
+            '--tagged-networks',
+            default=[],
+            dest='tagged_networks',
+            action='append',
+            metavar='<tagged_networks',
+            help=_("Name or UUID of tagged network")
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        tagged_networks = parsed_args.tagged_networks
+
+        if len(tagged_networks) == 0:
+            raise exceptions.CommandError(
+                "ERROR: no networks specified")
+
+        neutron_client = self.app.client_manager.network
+        trunk = neutron_client.find_trunk(parsed_args.name)
```

Comment:
What happens if this fails (e.g., operator provided an invalid trunk name on the command line)?

Reply:
Hm, it returns a weird error message. I'll catch this case.

Note that adding a network to a trunk where that network is already attached will result in a neutron error that's passed down appropriately, so I don't handle those cases. I guess it's just find_* methods that can return None and mess things up.

Reply:
Okay, this should be fixed now!